src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -787,6 +787,12 @@ void PolicyHandler::OnSystemRequestReceived() const {
   policy_manager_->ResetTimeout();
 }
 
+void PolicyHandler::TriggerPTUIfRequired() {
+#ifndef EXTERNAL_PROPRIETARY_MODE
+  policy_manager_->TriggerPTUIfRequired();
```

Comment:
I think the policy mode check should go where the call is originally triggered, at the end of `ApplicationManagerImpl::OnHMIStartedCooperation`

Reply:
The `TriggerPTUIfRequired` function is not implemented for external policies. In this case, it would be fine if I have the `EXTERNAL_POLICY` flag check in OnHMIStartedCooperation but If I had to use it somewhere else I’d have to add the policy flag check for each call. It might be neater for this to be handled in the policy handler. 

For now I think we could  add a note in the policy_handler function brief stating that its only implemented for regular policies at the moment

Reply:
Made the change in https://github.com/smartdevicelink/sdl_core/pull/3363/commits/454e5fbda47c6a7c0c14c8d8206354a33e1a3685